solution1.py

```diff
@@ -0,0 +1,32 @@
+# https://leetcode.com/problems/coin-change/
+
+# // Time Complexity : o(row*col) , row = number of coins + 1, col = amount+1
+# // Space Complexity : o(row*col) , row = number of coins + 1, col = amount+1
+# // Did this code successfully run on Leetcode :yes
+# // Any problem you faced while coding this : I got so lost in the indexing....
```

Comment:
Please do try to make the DP array and dry run your code on a whiteboard, that should help understanding and aide in visualization of indexing and navigating through the matrix